test/files/python_missing_check.expected

```diff
@@ -0,0 +1,22 @@
+Failure in python_missing_check.py:
+
+  The CHECKERR on line 7 wants:
+    BETA
+
+  which failed to match line stderr:1:
+    ALPHA
+
+  Context:
+    ALPHA (nothing to match)
+    BETA
+    GAMMA1 (nothing to match)
```

Comment:
I'm not sold on all these `(nothing to match)`. They seem a bit much, but like I said I don't want to rely on color, so we'd have to find some kind of text.

(also possible are leading symbols, or to switch the entire thing to unified diff format)

Reply:
Or possibly just print "(no CHECKs left)" once and then omit it since nothing after *could* match?

Reply:
I think it's fine, although printing the whole list might be a bit much.

Reply:
This does the thing that `diff -u` also does, which is to only show X lines of context for every mismatch (with X currently being hardcoded as 3)

So the only way this might print a lot is if there is a lot of output that doesn't have corresponding CHECKs?

I'm guessing in that case it depends on why there was that much more output? In case of stderr we already know that it might e.g. be asan output.

Do we truncate that output at e.g. 5 lines? Is that configurable? Or do we provide a logfile with the full output?

Truncating of course means that you might waste a run, and in case something is tough to reproduce that might be quite annoying, so I'd lean on the side of caution and give more output than strictly necessary.

Reply:
I like diff -u, but this the current version with "nothing to match" is really easy to read.

Maybe hardcode a context of 3-5 lines, so we don't print too many matching lines.
Mismatched lines should probably all be printed.